RadixWallet/Core/FeaturePrelude/Loadable.swift

```diff
@@ -258,6 +274,67 @@ extension Loadable {
 			.failure(error)
 		}
 	}
+
+    public func flatten<InnerValue>() -> Loadable<InnerValue> where Value == Loadable<InnerValue> {
+        switch self {
+        case .idle:
+            return .idle
+        case .loading:
+            return .loading
+        case let .success(value):
+            return value
+        case .failure(let error):
+            return .failure(error)
+        }
+    }
+
+    public func reduce(_ other: Loadable<Value>, join: (Value, Value) -> Value) -> Loadable<Value> {
+        concat(other).map(join)
+    }
+
+    public mutating func mutateValue(_ mutate: (inout Value) -> Void) {
+        switch self {
+        case .idle, .loading, .failure:
+            return
+        case var .success(value):
+            mutate(&value)
+            self = .success(value)
+        }
+    }
+
+    /// Refreshes from other Loadable by taking into account the current `success` state.
+    /// This is meant to preserve the `success` state while other Loadable is `loading` or `failed`.
+    public mutating func refresh(
```

Comment:
this is used onPullToRefresh event to facilitate the following - if the value is already loaded, on pull to refresh do not show the shimmer if the value is loading, or don't hide the value if new refresh failed. So the old value, if it was success, will be refreshed only if the new load result is also success.